vod/hls/m3u8_builder.c

```diff
@@ -565,78 +573,89 @@ m3u8_builder_build_master_playlist(
 	}
 
 	// write the header
-	p = vod_copy(result->data, m3u8_header, sizeof(m3u8_header) - 1);
 
+	p = vod_copy(result->data, m3u8_header, sizeof(m3u8_header) - 1);
 	// write the streams
 	for (cur_sequence = media_set->sequences; cur_sequence < media_set->sequences_end; cur_sequence++)
 	{
-		cur_sequence_tracks = cur_sequence->filtered_clips[0].longest_track;
-
-		// write the track information
-		if (cur_sequence_tracks[MEDIA_TYPE_VIDEO] != NULL)
-		{
-			track = cur_sequence_tracks[MEDIA_TYPE_VIDEO];
-			video = &track->media_info;
-			bitrate = video->bitrate;
-			if (cur_sequence_tracks[MEDIA_TYPE_AUDIO] != NULL)
+		main_media_type = cur_sequence->track_count[MEDIA_TYPE_VIDEO] != 0 ? MEDIA_TYPE_VIDEO : MEDIA_TYPE_AUDIO;
+             
+             
+		cur_sequence_tracks[MEDIA_TYPE_AUDIO] = cur_sequence->filtered_clips[0].longest_track[MEDIA_TYPE_AUDIO];
```

Comment:
looking at the updated code, we no longer need the cur_sequence_tracks array, we can instead add a new variable for the audio_track and use cur_track for the video. so basically:
cur_sequence_tracks[MEDIA_TYPE_VIDEO] != NULL becomes main_media_type == MEDIA_TYPE_VIDEO
cur_sequence_tracks[MEDIA_TYPE_VIDEO] becomes cur_track
cur_sequence_tracks[MEDIA_TYPE_AUDIO] becomes audio_track